qubekit/fragmentation/WBOFragmenter.py

```diff
@@ -0,0 +1,228 @@
+        ["[!#1]~[!$(*#*)&!D1:1]-,=;!@[!$(*#*)&!D1:2]~[!#1]"],
+        description="The rotatable smirks to use for fragmentation",
+    )
+
+    @classmethod
+    def is_available(cls) -> bool:
+        fragmenter = which_import(
+            "openff.fragmenter",
+            return_bool=True,
+            raise_error=True,
+            raise_msg="Please install via `conda install -c conda-forge openff-fragmenter`.",
+        )
+
+        rdkit = which_import(
+            "rdkit",
+            return_bool=True,
+            raise_error=True,
+            raise_msg="Please install via `conda install -c conda-forge rdkit`.",
+        )
+
+        return fragmenter and rdkit
+
+    def _prepare_fragment_input(self, molecule: Ligand):
+        """
+        TODO
+        """
+        # convert to OFF molecule
+        mapped_smiles = molecule.to_smiles(mapped=True)
+        off_molecule: OFFMolecule = OFFMolecule.from_mapped_smiles(mapped_smiles)
+
+        # remember the atom names
+        for off_atom, lig_atom in zip(off_molecule.atoms, molecule.atoms):
+            assert off_atom.atomic_number == lig_atom.atomic_number
+            off_atom.name = lig_atom.atom_name
+
+        # convert to the canonical order which will be used by the fragmenter
+        return off_molecule.canonical_order_atoms()
+
+    def _create_ligand(self, semi_mapped_smile, name, bond_indices=None):
+        """
+        TODO
+        """
+        # sanitize is off to keep the "atom_map" property
+        rdmol = Chem.MolFromSmiles(semi_mapped_smile, sanitize=False)
+
+        # add at least one conformer
+        Chem.AllChem.EmbedMolecule(rdmol)
+
+        # Chem.SanitizeMol calls updatePropertyCache so we don't need to call it ourselves
+        # https://www.rdkit.org/docs/cppapi/namespaceRDKit_1_1MolOps.html#a8d831787aaf2d65d9920c37b25b476f5
+        Chem.SanitizeMol(
+            rdmol,
+            Chem.SANITIZE_ALL ^ Chem.SANITIZE_ADJUSTHS ^ Chem.SANITIZE_SETAROMATICITY,
+        )
+        Chem.SetAromaticity(rdmol, Chem.AromaticityModel.AROMATICITY_MDL)
+
+        # Chem.MolFromSmiles adds bond directions (i.e. ENDDOWNRIGHT/ENDUPRIGHT), but
+        # doesn't set bond.GetStereo(). We need to call AssignStereochemistry for that.
+        Chem.AssignStereochemistry(rdmol)
+
+        # convert the rdmol to the qubekit Ligand
+        molecule = Ligand.from_rdkit(rdmol, name)
+
+        # attach the fragments as their own ligands
+        if bond_indices is not None:
+            molecule.bond_indices = bond_indices
+
+        # ensure that the atoms are ordered the same way
+        for atom, rdkit_atom in zip(molecule.atoms, rdmol.GetAtoms()):
+            if not rdkit_atom.GetAtomicNum() == atom.atomic_number:
+                raise Exception(f'Ligand created from an rdkit molecule should have the same atom order: rdkit {rdkit_atom.GetAtomMapNum()}, ligand {atom.atomic_number}')
+            atom.map_index = rdkit_atom.GetAtomMapNum()
```

Comment:
This should be moved the the `from_rdkit` method on the `Atom` class [here](https://github.com/qubekit/QUBEKit/blob/0c3c85de523e44f8dc32876c6b86e79d0479904f/qubekit/molecules/components.py#L162). We also need to check if the map is 0 as this indicates there is no map index in rdkit, this should then be stored as None in our model to clearly state there is no map.